src/openms/source/QC/MQEvidenceExporter.cpp

```diff
@@ -0,0 +1,406 @@
+        file_ << "NA" << "\t"; // Uncalibrated - Calibrated m/z [mDa]
+
+
+    }
+    else if (isnan(uncalibrated_mz_error_ppm))
+    {
+        file_ << calibrated_mz_error_ppm << "\t"; // Mass error [ppm]
+        file_ << "NA" << "\t"; // Uncalibrated Mass error [ppm]
+        file_ << OpenMS::Math::ppmToMass(calibrated_mz_error_ppm, f.getMZ()) << "\t"; // Mass error [Da]
+        file_ << "NA" << "\t"; // Uncalibrated Mass error [Da]
+        file_ << "NA" << "\t"; // Uncalibrated - Calibrated m/z [ppm]
+        file_ << "NA" << "\t"; // Uncalibrated - Calibrated m/z [mDa]
+    }
+    else
+    {
+        file_ << calibrated_mz_error_ppm << "\t";   //Mass error [ppm]
+        file_ << uncalibrated_mz_error_ppm << "\t"; // Uncalibrated Mass error [ppm]
+        file_ << OpenMS::Math::ppmToMass(calibrated_mz_error_ppm, f.getMZ()) << "\t"; // Mass error [Da]
+        file_ << OpenMS::Math::ppmToMass(uncalibrated_mz_error_ppm, f.getMZ()) << "\t"; // Uncalibrated Mass error [Da]
+        file_ << uncalibrated_mz_error_ppm - calibrated_mz_error_ppm << "\t"; // Uncalibrated - Calibrated m/z [ppm]
+        file_ << OpenMS::Math::ppmToMass((uncalibrated_mz_error_ppm - calibrated_mz_error_ppm), f.getMZ())
+              << "\t"; // Uncalibrated - Calibrated m/z [Da]
+    }
+    f.metaValueExists("rt_align_start") ? file_ << double(f.getMetaValue("rt_align_start")) / 60 << "\t" : file_ << "NA"
+                                                                                                                 << "\t"; //  Calibrated retention time start
+    f.metaValueExists("rt_align_end") ? file_ << double(f.getMetaValue("rt_align_end")) / 60 << "\t" : file_ << "NA"
+                                                                                                             << "\t"; // Calibrated retention time end
+    if (f.metaValueExists("rt_align"))
+    {
+        file_ << double(f.getMetaValue("rt_align")) / 60 << "\t"; // Calibrated Retention Time
+        file_ << (f.getRT() - double(f.getMetaValue("rt_align"))) / 60 << "\t"; // Retention time calibration
+    }
+    else
+    {
+        file_ << "NA" << "\t"; // calibrated retention time
+        file_ << "NA" << "\t"; // Retention time calibration
+    }
+    file_ << pep_ids_size << "\t"; // MS/MS count
+    if (type == "MULTI-MSMS")
+    {
+        file_ << "NA" << "\t"; // Match time diff
+        file_ << "NA" << "\t"; // Match mz diff
+    }
+    else
+    {
+        f.metaValueExists("rt_align") ? file_ << double(f.getMetaValue("rt_align")) - cmap[c_feature_number].getRT()
+                                              << "\t" : file_ << "NA" << "\t"; // Match time diff
+        file_ << f.getMZ() - cmap[c_feature_number].getMZ() << "\t";    //Match mz diff
+    }
+    file_ << raw_file << "\n"; // Raw File
+}
+
+void MQEvidence::exportFeatureMap(
+        const FeatureMap &feature_map,
+        const ConsensusMap &cmap) {
+    if (!isValid())
```

Comment:
why is exception only thrown here not in constructor?

Reply:
see above (comment to exception in constructor)